test/axiom/deploy_metachain_proxies.js

```diff
@@ -0,0 +1,143 @@
+  let consensus;
+
+  beforeEach(async () => {
+    config = {
+      techGov: accountProvider.get(),
+      consensusMasterCopy: await SpyConsensus.new(),
+      coreMasterCopy: await SpyCore.new(),
+      committeeMasterCopy: accountProvider.get(),
+      reputationMasterCopy: accountProvider.get(),
+      anchorMasterCopy: await SpyAnchor.new(),
+      consensusGatewayMasterCopy: await SpyConsensusGateway.new(),
+      txOptions: {
+        from: accountProvider.get(),
+      },
+    };
+
+    axiomContract = await AxiomUtils.deployAxiom(
+      config.techGov,
+      config.consensusMasterCopy.address,
+      config.coreMasterCopy.address,
+      config.committeeMasterCopy,
+      config.reputationMasterCopy,
+      config.anchorMasterCopy.address,
+      config.consensusGatewayMasterCopy.address,
+      {
+        from: accountProvider.get(),
+      },
+    );
+    const newCoreParams = {
+      consensus: accountProvider.get(),
+      metachainId: Utils.getRandomHash(),
+      epochLength: new BN('100'),
+      minValidators: new BN('10'),
+      joinLimit: new BN('10'),
+      reputation: accountProvider.get(),
+      height: new BN(Utils.getRandomNumber(1000)),
+      parent: Utils.getRandomHash(),
+      gasTarget: new BN(Utils.getRandomNumber(999999)),
+      dynasty: new BN(Utils.getRandomNumber(10)),
+      accumulatedGas: new BN(Utils.getRandomNumber(999999)),
+      source: Utils.getRandomHash(),
+      sourceBlockHeight: new BN(Utils.getRandomNumber(1000)),
+    };
+
+    coreSetupdata = await AxiomUtils.encodeNewCoreParams(newCoreParams);
+    anchorSetupData = await AxiomUtils.encodeNewAnchorParams({
+      maxStateRoots: new BN('100'),
+      consensus: accountProvider.get(),
+    });
+    consensusGatewaySetupData = await AxiomUtils.encodeNewConsensusGatewayParam();
+
+    consensus = newCoreParams.consensus;
+    await axiomContract.setConsensus(newCoreParams.consensus);
+  });
+
+  contract('Positive Tests', () => {
+    it('should deploy metachain proxies', async () => {
+      const returnValues = await axiomContract.deployMetachainProxies.call(
+        anchorSetupData,
+        coreSetupdata,
+        consensusGatewaySetupData,
+        { from: consensus },
+      );
+      const response = await axiomContract.deployMetachainProxies(
+        anchorSetupData,
+        coreSetupdata,
+        consensusGatewaySetupData,
+        { from: consensus },
+      );
+
+      assert.isOk(
+        response.receipt.status,
+        'Transaction should success',
+      );
+      assert.isOk(
+        web3.utils.isAddress(returnValues.anchor_),
+        'It should return valid anchor address',
+      );
+      assert.isOk(
+        web3.utils.isAddress(returnValues.core_),
+        'It should return valid core address',
+      );
+      assert.isOk(
+        web3.utils.isAddress(returnValues.consensusGateway_),
+        'It should return valid consensus gateway address',
+      );
+
+      assert.strictEqual(
```

Comment:
For all the deployed contracts, just check that the code length is greater than 2. 
Don't check the entire code.